src/Core/src/Test/ResultMockFactory.php

```diff
@@ -89,7 +89,11 @@ public static function create(string $class, array $data = [])
 
         $initializedProperty->setValue($object, true);
         foreach ($data as $propertyName => $propertyValue) {
-            $property = $reflectionClass->getProperty($propertyName);
+            if ($reflectionClass->hasProperty($propertyName)) {
+                $property = $reflectionClass->getProperty($propertyName);
+            } else {
+                $property = $reflectionClass->getProperty(lcfirst($propertyName));
```

Comment:
Could you add a comment before this line that this is legacy code?

Reply:
done